theano/sandbox/cuda/basic_ops.py

```diff
@@ -2437,6 +2437,102 @@ def perform(self, node, inp, out_):
 
         out[0] = x.reshape(tuple(shp))
 
+    def c_code_cache_version(self):
+        return (1,)
+
+    def c_code(self, node, name, inputs, outputs, sub):
+        x, shape = inputs
+        output, = outputs
+        new_ndim = self.ndim
+        sdtype = node.inputs[1].type.dtype_specs()[1]
+        fail = sub['fail']
+        return """
+        PyObject *new_shape = PyTuple_New(%(new_ndim)s);
```

Comment:
Just to let know this object wasn't freed. So there was a memory leak on the CPU. This is fixed by gh-3193